Lets Do This/Controllers/Settings/LDTSettingsViewController.m

```diff
@@ -157,14 +157,17 @@ - (void)handleLogoutTap:(UITapGestureRecognizer *)recognizer {
 
         [[DSOUserManager sharedInstance] endSessionWithCompletionHandler:^ {
             [SVProgressHUD dismiss];
-            [self.navigationController pushViewController:[[LDTUserConnectViewController alloc] init] animated:YES];
-            [self.navigationController styleNavigationBar:LDTNavigationBarStyleClear];
-            // Now that tabBar is hidden, select the first tab, so it will be the first tab selected upon next login.
-            LDTTabBarController *tabBar = (LDTTabBarController *)[[[[UIApplication sharedApplication] delegate] window] rootViewController];
-            [tabBar setSelectedIndex:0];
+            [self pushUserConnectViewController];
         } errorHandler:^(NSError *error) {
             [SVProgressHUD dismiss];
-            [LDTMessage displayErrorMessageForError:error];
+            // Performs normal logout functionality even if we are presented with an error,
+            // but only if error is not a lack of connectivity. 
+            if (error.code != -1009) {
+                [self pushUserConnectViewController];
+            }
+            else {
+                [LDTMessage displayErrorMessageForError:error];
```

Comment:
@tongxiang Did you test this? I've just tested and the `LDTMessage` is not visible. This method displays the error in the defaultViewController, which is the TabBarController, which sits underneath the presented Settings controller (since it's pushed modally)

This is caused by changes introduced in #633 -- which allows us to avoid constantly setting the `LDTMessage defaultViewController`.  The fix here is to use the new method `displayErrorInViewController:error`, displaying the message in `self.navigationController`.
This needs to use the